### ISSUE TITLE
feat(cast): add optimize selector command

### ISSUE THREAD

**Issue by ecmendenhall** [foundry-rs/foundry#2175]
### Component

Cast

### Describe the feature you would like

This [handy utility](https://emn178.github.io/solidity-optimize-name/) generates an optimized function name (i.e. a function selector with leading zero bytes) for a given Solidity function signature. For example, `transfer(address,uint256)` returns the optimized name `transfer_G8l(address,uint256)`, with the selector `0x0000c771`. 

This would be a nice helper utility to add to `cast`.

Seems like this might make sense as a flag or subcommand under `cast sig`. Maybe something like:

```bash
$ cast sig "transfer(address,uint256)"
0xa9059cbb
$ cast sig "transfer(address,uint256)" --optimize
transfer_G8l(address,uint256)
0x0000c771
```

### Additional context

(Inspired by a deleted tweet from @transmissions11)

**Comment by ecmendenhall** [comment]
(I'd be up for adding this myself once there's agreement on the CLI interface)

**Comment by onbjerg** [comment]
Interface looks good to me but pinging @transmissions11 for feedback

**Comment by transmissions11** [comment]
interface looks good, i like putting it with sig, only request would be a flag that lets you configure how many zeros you want to target (the website won't add more than 3 random chars after the sig to produce zeros)

**Comment by Evalir** [comment]
Closed with #4705
